### ISSUE TITLE
Equal rules don't merge if exists a different among them

### ISSUE THREAD

**Issue by asfixia** [css/csso#394]
The rules (in input):

CSS first (with problem):

> .x-btn-group-tl {
> 	background-image: url(../images/default/button/group-cs.gif);
> }
> 
> .x-btn-group-tr {
> 	background-image: url(../images/default/button/group-cs.gif);
> }
> 
> .x-btn-group-bc {
> 	background-image: url(../images/default/button/group-tb.gif);
> }
> 
> .x-btn-group-bl {
> 	background-image: url(../images/default/button/group-cs.gif);
> }
> 
> .x-btn-group-br {
> 	background-image: url(../images/default/button/group-cs.gif);
> }

Gives from first:

> .x-btn-group-tl,.x-btn-group-tr{background-image:url(../images/default/button/group-cs.gif)}.x-btn-group-bc{background-image:url(../images/default/button/group-tb.gif)}.x-btn-group-bl,.x-btn-group-br{background-image:url(../images/default/button/group-cs.gif)}

CSS second (work as expected):

> .x-btn-group-tl {
> 	background-image: url(../images/default/button/group-cs.gif);
> }
> 
> .x-btn-group-tr {
> 	background-image: url(../images/default/button/group-cs.gif);
> }
> 
> .x-btn-group-bl {
> 	background-image: url(../images/default/button/group-cs.gif);
> }
> 
> .x-btn-group-br {
> 	background-image: url(../images/default/button/group-cs.gif);
> }

Gives 2:

> .x-btn-group-bl,.x-btn-group-br,.x-btn-group-tl,.x-btn-group-tr{background-image:url(../images/default/button/group-cs.gif)}

Both results should be equals, isnt?

**Comment by kizu** [comment]
This is an expected behaviour. The order of rules in CSS matters, so, for example, if you'd have two following elements:

``` html
<div class="x-btn-group-tr x-btn-group-bc"></div>
<div class="x-btn-group-bc x-btn-group-bl"></div>
```

For the unminified version the first element would have `group-tb.gif` as a background, and the second would have `group-cs.gif`.

If CSSO would merge all the rules with `group-cs.gif` into one, then it would be placed either before the one with `group-tb.gif` or after, and both elements would have the same background applied, so there would be a change in the behaviour.

Basically, CSSO doesn't know how your CSS would be applied, so it can't move stuff like that.

However, there is a `scopes` option which you can use to tell CSSO which classnames could be merged like you want: https://github.com/css/csso#scopes — you can try to utilise it.

**Comment by asfixia** [comment]
Yeah, @kizu  is right.
Thanks!
